app/models/email_change_request.rb

```diff
@@ -4,30 +4,40 @@ class EmailChangeRequest < ActiveRecord::Base
   belongs_to :old_email_token, class_name: 'EmailToken'
   belongs_to :new_email_token, class_name: 'EmailToken'
   belongs_to :user
+  belongs_to :requested_by, class_name: "User", foreign_key: :requested_by_user_id
 
   validates :new_email, presence: true, format: { with: EmailValidator.email_regex }
 
   def self.states
     @states ||= Enum.new(authorizing_old: 1, authorizing_new: 2, complete: 3)
   end
 
+  def requested_by_admin?
```

Comment:
This is misleading because the property is `staff?` which can be true for admins or moderators.

I recommend renaming it to `requested_by_staff?`

Reply:
This should also use `&& ! self_requested?` -- an administrator changing their own email shouldn't get the special "was requested by someone else" message.